core/exchange.go

```diff
@@ -173,12 +173,12 @@ func (ce *Exchange) getExtendedHeaderByHeight(ctx context.Context, height *int64
 	}
 	log.Debugw("fetched signed block from core", "height", b.Header.Height)
 
-	eds, err := extendBlock(b.Data, b.Header.Version.App)
+	eds, err := extendBlock(*b.Data, b.Header.Version.App)
```

Comment:
this will be GC at some point. The thing is `extendBlock` also makes a couple other calls by value instead of reference. So multiple copies will be done regardless, this will be just one less. What do you think of creating an issue that checks wherever `Data` is used, it passes it by reference, even in app and maybe other repos so that we have everything fixed?

Reply:
That's also fine, but `extendBlock` used to accept data by ref, so why make this change here?

Reply:
Why not both? (pass by ref here and also open an issue to change other calls to accept data by ref)